pkg/controller/operators/olm/operator.go

```diff
@@ -310,6 +314,32 @@ func (a *Operator) syncObject(obj interface{}) (syncError error) {
 		a.requeueOwnerCSVs(metaObj)
 	}
 
+	namespace, ok := obj.(*corev1.Namespace)
+	if ok {
+		// Check to see if any operator groups are associated with this namespace
+		operatorGroupList, err := a.lister.OperatorsV1alpha2().OperatorGroupLister().OperatorGroups(metav1.NamespaceAll).List(labels.Everything())
+		if err != nil {
+			syncError = fmt.Errorf("lister failed: %v", err)
+			logger.Warn(syncError.Error())
+			return
+		}
+		for _, opGroup := range operatorGroupList {
+			namespaceMap, err := a.getMatchingNamespaces(opGroup)
```

Comment:
We shouldn't need to list all of the namespaces of the operator group to know if the one we're currently syncing should requeue an operatorgroup? Just need to compare the namespace list (in spec of opgroup) or the label selector on the operatorgroup / labels of the namespace.

We probably already have a function somewhere that does this, but don't we just need to do something like `if ShouldContain(opGroup, namespace) { resync opgroup }`?

Reply:
The resyncing here is specifically to make an operator group status contain a properly updated namespace list. You are right though that less work is necessary here, so I'll look into fixing that.

Reply:
I take back what I said about less work being necessary. I think it should be left as is.